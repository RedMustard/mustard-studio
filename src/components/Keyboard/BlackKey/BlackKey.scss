@import '../../../style/variables.scss';

.keyboard-key--black {
    background-color: #000;
    border: none;
    box-shadow: 0 2px 5px #999;
    display: table-cell;
    height: $keyboardHeight * 2 / 3;
    outline: none;
    position: absolute;
    transform: translateX(-12px);
    width: $whiteKeyWidth * .54;

    &:hover {
        background-color: #444;
    }

    &:active {
        background-color: #333;
    }
}
