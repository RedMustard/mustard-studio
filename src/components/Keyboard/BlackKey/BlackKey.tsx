import { h } from 'preact';
import { KeyboardKey } from '../KeyboardKey/KeyboardKey';

interface BlackKeyProps {
    onMouseLeave: (e: MouseEvent) => void;
    onMouseOver: (e: MouseEvent) => void;
    onMouseDown: (e: MouseEvent) => void;
    onMouseUp: (e: MouseEvent) => void;
}

export const BlackKey = ({
    onMouseLeave,
    onMouseOver,
    onMouseDown,
    onMouseUp,
}: BlackKeyProps) => (
    <KeyboardKey
        className="keyboard-key--black"
        onMouseLeave={(e: MouseEvent) => onMouseLeave(e)}
        onMouseOver={(e: MouseEvent) => onMouseOver(e)}
        onMouseDown={(e: MouseEvent) => onMouseDown(e)}
        onMouseUp={(e: MouseEvent) => onMouseUp(e)}
    />
);
