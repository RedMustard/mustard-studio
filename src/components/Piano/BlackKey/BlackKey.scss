@import '../../../style/variables.scss';

.piano-key--black {
    background-color: #151515;
    border: none;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    box-shadow: 0 1px 3px #222;
    display: table-cell;
    height: $pianoHeight * 2 / 3;
    outline: none;
    position: absolute;
    transform: translateX(-12px);
    width: $whiteKeyWidth * .54;

    &:hover {
        background-color: #222;
    }

    &:active {
        background-color: #111;
    }
}
