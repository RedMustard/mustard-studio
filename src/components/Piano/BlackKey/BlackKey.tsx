import { h } from 'preact';
import { PianoKey } from '../PianoKey/PianoKey';

interface BlackKeyProps {
    onMouseLeave: (e: MouseEvent) => void;
    onMouseOver: (e: MouseEvent) => void;
    onMouseDown: (e: MouseEvent) => void;
    onMouseUp: (e: MouseEvent) => void;
}

export const BlackKey = ({
    onMouseLeave,
    onMouseOver,
    onMouseDown,
    onMouseUp,
}: BlackKeyProps) => (
    <PianoKey
        className="piano-key--black"
        onMouseLeave={(e: MouseEvent) => onMouseLeave(e)}
        onMouseOver={(e: MouseEvent) => onMouseOver(e)}
        onMouseDown={(e: MouseEvent) => onMouseDown(e)}
        onMouseUp={(e: MouseEvent) => onMouseUp(e)}
    />
);
