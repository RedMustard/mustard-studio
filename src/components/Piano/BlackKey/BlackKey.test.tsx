import { h } from 'preact';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BlackKey } from './BlackKey';


const baseProps = {
    onMouseDown: jest.fn(),
    onMouseUp: jest.fn(),
    onMouseOver: jest.fn(),
    onMouseLeave: jest.fn(),
};

describe('<BlackKey />', () => {
    it('renders with basic props', () => {
        const wrapper = shallow(
            <BlackKey
                {...baseProps}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('fires onMouseDown', () => {
        const wrapper = mount(
            <BlackKey
                {...baseProps}
            />,
        );
        wrapper.find('div').simulate('mousedown');
        expect(baseProps.onMouseDown).toBeCalled();
    });

    it('fires onMouseUp', () => {
        const wrapper = mount(
            <BlackKey
                {...baseProps}
            />,
        );
        wrapper.find('div').simulate('mouseup');
        expect(baseProps.onMouseUp).toBeCalled();
    });

    it('fires onMouseOver', () => {
        const wrapper = mount(
            <BlackKey
                {...baseProps}
            />,
        );
        wrapper.find('div').simulate('mouseover');
        expect(baseProps.onMouseDown).toBeCalled();
    });

    it('fires onMouseLeave', () => {
        const wrapper = mount(
            <BlackKey
                {...baseProps}
            />,
        );
        wrapper.find('div').simulate('mouseleave');
        expect(baseProps.onMouseDown).toBeCalled();
    });
});
