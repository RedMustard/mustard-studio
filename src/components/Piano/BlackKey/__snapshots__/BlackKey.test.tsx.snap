// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlackKey /> renders with basic props 1`] = `
<PianoKey
  className="piano-key--black"
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
/>
`;
